app/controllers/teachers/students_controller.rb

```diff
@@ -1,16 +1,73 @@
 module Teachers
   class StudentsController < ApplicationController
     before_action :authenticate_user!, :set_user
+    before_action :set_student, only: [:show, :edit, :update, :destroy]
 
     def index
       @students = User.find(@user.group_members.map(&:user_id))
     end
 
+    def show; end
+
+    def new
+      @student = User.new
+    end
+
+    def edit; end
+
+    def create
+      ActiveRecord::Base.transaction do
+        @student = User.create(user_params.merge(password: new_user_temp_password, role: "student"))
+        StudentInformation.create(student_information_params.merge(user_id: @student.id))
+        GroupMember.create(group_id: params[:user][:group], user: @student)
+      end
+
+      return redirect_to_index if @student.save
+
+      render :new, status: :unprocessable_entity
+    end
+
+    def update
+      ActiveRecord::Base.transaction do
+        @student.update(user_params)
```

Comment:
I go the feeling this three updates could be perform in a single action
You may need [nested attributes ](https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html)
Also see [accepts_nested_attributes_for](https://apidock.com/rails/ActiveRecord/NestedAttributes/ClassMethods/accepts_nested_attributes_for)

Reply:
A powerful tool nested_attributes! Fixed.